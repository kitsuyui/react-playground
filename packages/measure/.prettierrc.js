module.exports = {
  ...require('@kitsuyui/prettier-config'),
}
