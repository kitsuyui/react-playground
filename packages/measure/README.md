# @kitsuyui/react-measure

[![npm version](https://badge.fury.io/js/@kitsuyui%2Freact-measure.svg)](https://badge.fury.io/js/@kitsuyui%2Freact-measure)

React component that gets the size of the parent element and renders it.

## Demo

Storybook: https://react-playground.docs.kitsuyui.com/storybook/

## Installation

### npm

```sh
npm install @kitsuyui/react-measure
```

### yarn

```sh
yarn add @kitsuyui/react-measure
```

### pnpm

```sh
pnpm add @kitsuyui/react-measure
```

## License

MIT
