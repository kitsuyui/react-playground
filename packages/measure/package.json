{
  "name": "@kitsuyui/react-measure",
  "version": "0.0.0",
  "files": [
    "dist",
    "package.json"
  ],
  "exports": {
    ".": {
      "require": "./dist/index.js",
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.ts"
    }
  },
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-use": "^17.4.0"
  },
  "scripts": {
    "build": "tsup",
    "dev": "tsup --watch",
    "test": "jest --watchAll=false --collect-coverage"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@swc/core": "^1.3.82",
    "@swc/jest": "^0.2.29",
    "@testing-library/jest-dom": "^6.1.2",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/jest": "^29.5.4",
    "@types/node": "^20.5.7",
    "@types/react": "^18.2.21",
    "@types/react-dom": "^18.2.7",
    "jest": "^29.6.4",
    "jest-environment-jsdom": "^29.6.4",
    "ts-node": "^10.9.1",
    "tsup": "^7.2.0",
    "typescript": "^5.2.2"
  },
  "license": "MIT",
  "author": "Yui Kitsu <kitsuyui@kitsuyui.com>",
  "description": "A react component to measure the size of a DOM node."
}
