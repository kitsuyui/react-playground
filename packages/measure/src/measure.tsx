import React from 'react'
import { createContext } from 'react'
import { useMeasure } from 'react-use'
import { MeasureStyle, MeasureBase, DEFAULT_MEASURE_STYLE } from './base'

export const MeasureStyleContext = createContext<MeasureStyle>(DEFAULT_MEASURE_STYLE)

/**
 * Measure component
 *
 * This component shows the size of the parent element.
 * @returns {JSX.Element}
 */
export const Measure: React.FC = (): JSX.Element => {
  const [ref, { width, height }] = useMeasure<HTMLDivElement>()
  const styles = React.useContext(MeasureStyleContext)
  return (
    <div
      ref={ref}
      style={{
        width: '100%',
        height: '100%',
        position: 'absolute',
        overflow: 'hidden',
      }}
    >
      <MeasureBase width={width} height={height} styles={styles} />
    </div>
  )
}
