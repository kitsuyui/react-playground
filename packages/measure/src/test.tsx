import { render } from '@testing-library/react'
import React from 'react'

import { Measure } from './measure'

test('render Measure', () => {
  const element = <Measure />
  render(
    <div
      style={{
        width: '500px',
        height: '400px',
        position: 'relative',
        overflow: 'hidden',
      }}
    >
      {element}
    </div>
  )
  expect(element).toMatchSnapshot()
})
