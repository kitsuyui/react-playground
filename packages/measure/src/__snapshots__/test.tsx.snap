// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Measure 1`] = `<Measure />`;
